### PR TITLE
Feature llm providers refactoring

### PR DESCRIPTION
**Pull request by Winston-503**
- Refactor `llm` package to introduce `providers` with subfolders for `anthropic`, `gemini`, `groq`, `ollama` and `openai` (for both OpenAI and Azure)
- Refactor consumption calculation logic in separate `{provider}_llm_cost.py` files and distribute `yaml` cost files
- Improve provider selection logic towards providers scalability by implementing general `from_env()` and `from_config()` for `LLMBase`
- Add unit tests for Groq cost calculations

### CODE REVIEW

File: council/llm/__init__.py [hunk 1]
```diff
@@ -43,30 +42,24 @@
...
-
-from .gemini_llm_configuration import GeminiLLMConfiguration
-from .gemini_llm import GeminiLLM
-
-from .ollama_llm_configuration import OllamaLLMConfiguration
-from .ollama_llm import OllamaLLM
-
-from .groq_llm_configuration import GroqLLMConfiguration
-from .groq_llm import GroqLLM
+from .providers import (
```
Comment (Winston-503) [inline council/llm/__init__.py:45]: Don't import low-level `ChatGPTConfigurationBase`, `OpenAIChatCompletionsModel` and `OpenAITokenCounter`  here by analogy to not importing `AnthropicCompletionLLM`, `AnthropicMessagesLLM` from `anthropic`

File: council/llm/llm_fallback.py [hunk 2]
```diff
@@ -19,6 +29,17 @@ def __init__(self, *, llm_config: T_Configuration, llm_fallback_config: T_Config
     def model_name(self) -> str:
         return f"{self._llm_config.model_name()} with fallback_{self._llm_fallback_config.model_name()}"
 
+    @classmethod
+    def from_env(cls, *args: Any, **kwargs: Any) -> LLMFallbackConfiguration:
+        raise ValueError("LLMFallbackConfiguration doesn't support from_env() initialization.")
```
Comment (aflament) [inline council/llm/llm_fallback.py:34]: I would have expected `NotImplementedError`

File: council/llm/providers/anthropic/anthropic_llm_configuration.py [hunk 3]
```diff
@@ -124,8 +124,11 @@ def from_env() -> AnthropicLLMConfiguration:
         config._read_optional_env()
         return config
 
-    @staticmethod
-    def from_spec(spec: LLMConfigSpec) -> AnthropicLLMConfiguration:
+    @classmethod
+    def from_spec(cls, spec: LLMConfigSpec) -> AnthropicLLMConfiguration:
+        if not spec.provider.is_of_kind(LLMProviders.Anthropic):
+            raise ValueError(f"Invalid LLM provider, actual {spec.provider}, expected {LLMProviders.Anthropic}")
+
```
Comment (aflament) [inline council/llm/providers/anthropic/anthropic_llm_configuration.py:131]: Add a `check_provider` in the base class or `LLMConfigSpec`

`def check_provider(provider: LLMProviders) -> None:
        if not spec.provider.is_of_kind(provider):
            raise ValueError(f"Invalid LLM provider, actual {spec.provider}, expected {provider}")
`

File: council/mocks/mock_llm.py [hunk 4]
```diff
@@ -43,6 +44,14 @@ def __init__(self, model_name: str, timeout: int = 10, token_limit: int = -1) ->
         self._timeout = timeout
         self._token_limit = token_limit
 
+    @classmethod
+    def from_env(cls, *args: Any, **kwargs: Any) -> MockLLMConfiguration:
+        raise ValueError("MockLLMConfiguration doesn't support from_env() initialization.")
```
Comment (aflament) [inline council/mocks/mock_llm.py:49]:
```suggestion
        raise NotImplementedError("MockLLMConfiguration doesn't support from_env() initialization.")
```

File: council/mocks/mock_llm.py [hunk 5]
```diff
@@ -43,6 +44,14 @@ def __init__(self, model_name: str, timeout: int = 10, token_limit: int = -1) ->
...
+
+    @classmethod
+    def from_spec(cls, spec: LLMConfigSpec) -> MockLLMConfiguration:
+        raise ValueError("MockLLMConfiguration doesn't support from_spec() initialization.")
```
Comment (aflament) [inline council/mocks/mock_llm.py:53]:
```suggestion
        raise NotImplementedError("MockLLMConfiguration doesn't support from_spec() initialization.")
```

**Review by aflament — APPROVED** [review]

**Pull request merged**